### PR TITLE
FormEngine chapter Cleanup

### PR DESCRIPTION
**Pull request by linawolf**
references #2782

### CODE REVIEW

File: Documentation/ApiOverview/FormEngine/DataCompiling/Index.rst [hunk 1]
```diff
@@ -172,42 +172,30 @@ it, :php:`TcaFlexProcess` uses the prepared data structure, processes values and
...
-        [\GeorgRinger\News\Backend\FormDataProvider\NewsFlexFormManipulation::class] = [
-            'depends' => [
-                \TYPO3\CMS\Backend\Form\FormDataProvider\TcaFlexPrepare::class,
-            ],
-            'before' => [
-                \TYPO3\CMS\Backend\Form\FormDataProvider\TcaFlexProcess::class,
-            ],
-        ];
-    }
+..  literalinclude:: _ext_localconf.php
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/DataCompiling/Index.rst:175]: Currently, this snippet is not used anymore in EXT:news. We can leave it, but add a hint that it is a former version of news we are referring to. What do you think?

https://github.com/georgringer/news/blob/main/ext_localconf.php#L104-L112

File: Documentation/ApiOverview/FormEngine/Rendering/Index.rst [hunk 2]
```diff
@@ -83,60 +71,39 @@ including containers, elements, fieldInformation, fieldWizards and fieldControls
...
-        'config' => [
-            'type' => 'select',
-            'renderType' => 'selectTagCloud',
-            'foreign_table' => 'tx_cooltagcloud_availableTags',
-        ],
-    ];
-
+..  literalinclude:: _tx_cooltagcloud.php
+    :language: php
+    :caption: EXT:cool_tag_cloud/Configuration/TCA/overrides/tx_cooltagcloud.php
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/Index.rst:106]: What is cool_tag_cloud? The extension does not exist on TER? I am fine with adjusting it in a separate PR.

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\FormEngine\FieldControl;
+
+use TYPO3\CMS\Backend\Form\AbstractNode;
+
+class ImportDataControl extends AbstractNode
+{
+    private string $langFile = 'LLL:EXT:my_extension/Ressources/private/Languages/locallang_db.xlf';
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:11]:
```suggestion
    private string $langFile = 'LLL:EXT:my_extension/Ressources/private/Languages/locallang_db.xlf';

```

For better readability: add an empty line.

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface $response
+     * @return ResponseInterface
+     */
+    public function importDataAction(ServerRequestInterface $request, ResponseInterface $response): ResponseInterface
+    {
+        $queryParameters = $request->getParsedBody();
+        $id = (int)$queryParameters['id'];
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php:20]:
```suggestion
        $id = (int)($queryParameters['id'] ?? 0);
```

Maybe we may run into a warning when not having a fallback.

File: Documentation/ApiOverview/FormEngine/Rendering/_InputTextElement.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        ],
+        'defaultLanguageDifferences' => [
+            'renderType' => 'defaultLanguageDifferences',
+            'after' => [
+                'otherLanguageContent',
+            ],
+        ],
+    ];
+
+    public function render()
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_InputTextElement.php:28]:
```suggestion
    public function render(): array
```

We can use covariance here for better type safety.

**Review by brotkrueml — CHANGES_REQUESTED** [review]
Some classes are final, some are not. Any particular reason?

File: Documentation/ApiOverview/FormEngine/DataCompiling/Index.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ApiOverview/FormEngine/DataCompiling/Index.rst:177]: Added a comment

**Comment by linawolf** [comment]
I made all classes final, had forgotten some

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\FormEngine\FieldControl;
+
+use TYPO3\CMS\Backend\Form\AbstractNode;
+
+final class ImportDataControl extends AbstractNode
+{
+    private string $langFile = 'LLL:EXT:my_extension/Ressources/private/Languages/locallang_db.xlf';
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:11]:
```suggestion
    private string $langFile = 'LLL:EXT:my_extension/Ressources/Private/Languages/locallang_db.xlf';
```

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+        if (empty($id)) {
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php:22]: No change and no response? Have you just overlooked it?

File: Documentation/ApiOverview/FormEngine/Rendering/_InputTextElement.php (hunk 5 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_InputTextElement.php:28]: No change and no response. Have you overlooked it?

File: Documentation/ApiOverview/FormEngine/Rendering/_ext_localconf_flex.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+declare(strict_types=1);
+
+use TYPO3\CMS\Compatibility6\Form\Container\FlexFormEntryContainer;
+
+defined('TYPO3') or die();
+
+// Default registration of "flex" in NodeFactory:
+// 'flex' => \TYPO3\CMS\Backend\Form\Container\FlexFormEntryContainer::class,
+
+// Register language aware flex form handling in FormEngine
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_ext_localconf_flex.php:12]: No change and no response. Have you overlooked it?

File: Documentation/ApiOverview/FormEngine/Rendering/_pages.php [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+defined('TYPO3') or die();
+
+(static function (): void {
+    $langFile = 'LLL:EXT:my_extension/Ressources/private/Languages/locallang.xlf';
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_pages.php:6]:
```suggestion
    $langFile = 'LLL:EXT:my_extension/Ressources/Private/Language/locallang.xlf';
```

File: Documentation/ApiOverview/FormEngine/Rendering/_tx_cooltagcloud.php [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+defined('TYPO3') or die();
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_tx_cooltagcloud.php:3]: No change and no response. Have you overlooked it?

File: Documentation/ApiOverview/FormEngine/Rendering/Index.rst [hunk 11]
```diff
@@ -148,16 +115,11 @@ that are called for node name "text", and if the TCA config enables the editor,
...
 
-    // Register FormEngine node type resolver hook to render RTE in FormEngine if enabled
-    $GLOBALS['TYPO3_CONF_VARS']['SYS']['formEngine']['nodeResolver'][1480314091] = [
-        'nodeName' => 'text',
-        'priority' => 50,
-        'class' => \TYPO3\CMS\RteCKEditor\Form\Resolver\RichTextNodeResolver::class,
-    ];
-
-The trick is here that "ckeditor" registers his resolver with ah higher priority (50) than "rtehtmlarea" (40), so the
+The trick here is that "ckeditor" registers his resolver with ah higher priority (50) than "rtehtmlarea" (40), so the
```
[the remark this answers is not in the document]
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/Index.rst:122]: No change and no response. Have you overlooked it?

File: Documentation/ApiOverview/FormEngine/Rendering/Index.rst [hunk 12]
```diff
@@ -83,60 +71,39 @@ including containers, elements, fieldInformation, fieldWizards and fieldControls
 can be fully adapted and extended if needed. It is possible to transparently "kick-out" a Core container and to
 substitute it with an own implementation.
 
-As example, the TemplaVoila implementation needs to add additional render capabilities of the flex form rendering
+For example, the TemplaVoila implementation needs to add additional render capabilities of the flex form rendering
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/Index.rst:74]:
```suggestion
For example, the TemplaVoila implementation needs to add additional render capabilities of the FlexForm rendering
```

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php (hunk 4 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_ImportDataController.php:20]: Ah sorry, found another one without response.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**